src/barnote.ts

```diff
@@ -43,34 +53,31 @@ export class BarNote extends Note {
   }
 
   // Get and set the type of Bar note. `type` must be one of `Vex.Flow.Barline.type`.
-  getType() {
+  getType(): number {
     return this.type;
   }
-  setType(type) {
+
+  setType(type: string | number): this {
```

Comment:
similarly instead of number can we put BarlineType?

Reply:
done